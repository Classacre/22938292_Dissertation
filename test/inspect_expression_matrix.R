# Inspect the expression matrix and GBM gene overlap
expr_mat <- as.matrix(read.csv("/group/sms029/mnieuwenh/gbm_noise_analysis/expression_matrix.csv", row.names=1, check.names=FALSE))
gbm_genes <- read.csv("/group/sms029/mnieuwenh/gbM_data/unique_gbm_genes.csv", stringsAsFactors=FALSE)$Gene

cat("Expression matrix dimensions: ", dim(expr_mat)[1], "genes x", dim(expr_mat)[2], "cells\n")
cat("First 5 gene names in matrix:\n")
print(head(rownames(expr_mat), 5))
cat("Number of unique GBM genes in list: ", length(unique(gbm_genes)), "\n")
cat("Number of genes in matrix that are GBM: ", sum(rownames(expr_mat) %in% gbm_genes), "\n")
cat("Number of genes in matrix that are non-GBM: ", sum(!(rownames(expr_mat) %in% gbm_genes)), "\n")
cat("Are there both GBM and non-GBM genes in the matrix? ", 
    (sum(rownames(expr_mat) %in% gbm_genes) > 0) & (sum(!(rownames(expr_mat) %in% gbm_genes)) > 0), "\n")
