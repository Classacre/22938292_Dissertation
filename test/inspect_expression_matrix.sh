#!/bin/bash
#SBATCH --job-name=inspect_expr_matrix
#SBATCH --output=/group/sms029/mnieuwenh/test/logs/inspect_expression_matrix.out
#SBATCH --error=/group/sms029/mnieuwenh/test/logs/inspect_expression_matrix.err
#SBATCH --partition=work
#SBATCH --mem=120G
#SBATCH --cpus-per-task=2
#SBATCH --time=00:30:00
#SBATCH --mail-user=22938292@student.uwa.edu.au
#SBATCH --mail-type=END,FAIL

module load Anaconda3/2024.06
source ~/.bashrc
conda activate /group/sms029/conda_environment/R

Rscript /group/sms029/mnieuwenh/test/inspect_expression_matrix.R
