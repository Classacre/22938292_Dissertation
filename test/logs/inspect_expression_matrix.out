Expression matrix dimensions:  25784 genes x 87056 cells
First 5 gene names in matrix:
[1] "1" "2" "3" "4" "5"
Number of unique GBM genes in list:  11272 
Number of genes in matrix that are GBM:  0 
Number of genes in matrix that are non-GBM:  25784 
Are there both GBM and non-GBM genes in the matrix?  FALSE 
